mac_installer/PostInstall.cpp

```diff
@@ -1220,6 +1224,95 @@ Boolean CheckDeleteFile(char *name)
     return false;
 }
 
+
+// If there are other copies of BOINC Manager with different branding
+// on the system, Noitifications may display the icon for the wrong
+// branding, due to the Launch Services database having one of the
+// other copies of BOINC Manager as the first entry. Each user has
+// their own copy of the Launch Services database, so this must be
+// done for each user.
+//
+// This probably will happen only on BOINC development systems where
+// Xcode has generated copies of BOINC Manager.
+static void FixLaunchServicesDataBase(uid_t userID, long brandID) {
+    uid_t saved_uid;
+    char boincPath[MAXPATHLEN];
+    char cmd[MAXPATHLEN+250];
+    long i, n;
+    CFArrayRef appRefs = NULL;
+    OSStatus err;
+
+    if (compareOSVersionTo(10, 8) < 0) {
+        return;  // Notifications before OS 10.8 just bounce our Dock icon
+    }
+
+    saved_uid = geteuid();
+    CFStringRef bundleID = CFSTR("edu.berkeley.boinc");
+
+    if (LSCopyApplicationURLsForBundleIdentifier) { // Weak linked; not available before OS 10.10
+        seteuid(userID);    // Temporarily set effective uid to this user
+        appRefs = LSCopyApplicationURLsForBundleIdentifier(bundleID, NULL);
+        seteuid(saved_uid);     // Set effective uid back to privileged user
+            if (appRefs == NULL) {
+                printf("Call to LSCopyApplicationURLsForBundleIdentifier returned NULL\n");
+                goto registerOurApp;
+            }
+            n = CFArrayGetCount(appRefs);   // Returns all results at once, in database order
+    } else {
```

Comment:
Could be moved outside of the else block before
`if (LSCopyApplicationURLsForBundleIdentifier) { // Weak linked; not available before OS 10.10`

Reply:
It could, but I think it is better as is, because it is closer to the for() statement, improving readability. It's really a matter of coding style and individual preference.

Reply:
Ok, I prefer to get variable initialized with the default value but I'm not forcing you or anyone else to change this piece of code according to my preferences :)